kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutlinePanel.js

```diff
@@ -0,0 +1,208 @@
+/* eslint-env browser */
+/**
+ * Widget for Narrative Outline Panel
+ * @author David Lyon  <dlyon@lbl.gov>
+ * @public
+ */
+define(['jquery', 'base/js/namespace', 'kbwidget', 'kbaseNarrativeControlPanel'], (
+    $,
+    Jupyter,
+    KBWidget,
+    ControlPanel
+) => {
+    'use strict';
+    return new KBWidget({
+        name: 'kbaseNarrativeOutlinePanel',
+        parent: ControlPanel,
+        version: '1.0.0',
+        options: {
+            /** If true, all markdown H1-H3 will be outline items
+             * If false, only the first header will be an outline item */
+            showAllHeaders: false,
+            /** Header height for visibility & scroll functionality */
+            headerHeight: 80,
+            /** Refresh throttle rate in ms */
+            refreshRate: 100,
+        },
+
+        isActive: false,
+        body: undefined,
+        drawnState: '', // string representing the items drawn used to eliminate excess dom manipulation
```

Comment:
Can you use a single comment type so it's easy for readers to parse the text?

Generally Bill and I go for
```js
    // a comment about a variable
    headerHeight: 80,
    // a multi-line comment about
    // a variable
    showAllHeaders: false,
    // another comment
    refreshRate: 100,
```

```js
   /**
    * method documentation
    * 
    * @param {string} whatever - response to comments
    */
    respond: function (whatever) {
        ...
```